collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -92,7 +93,36 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     private void initToolbar() {
         mToolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(mToolbar);
+        mToolbar.inflateMenu(R.menu.menu);
+
+        mToolbar.findViewById(R.id.menu_sort).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mDrawerLayout.isDrawerOpen(Gravity.END)) {
+                    mDrawerLayout.closeDrawer(Gravity.END);
+                } else {
+                    Collect.getInstance().hideKeyboard(mInputSearch);
+                    mDrawerLayout.openDrawer(Gravity.END);
+                }
+            }
+        });
+
+        mToolbar.findViewById(R.id.menu_filter).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSearchBoxLayout.getVisibility() == View.GONE) {
+                    showSearchBox();
+                } else {
+                    hideSearchBox();
+                }
+            }
+        });
+
+        boolean hasHardwareMenu =
```

Comment:
No, I couldn't find out the documentation as to why it wasn't working. My guess is that I have used a custom toolbar in all of the activities. So it isn't able to exhibit that behaviour on its own. 
As a solution to solve this problem, I have checked whether the device has menu button explicitly.
Is this correct?